eth_tester/utils/backend_testing.py

```diff
@@ -452,15 +453,15 @@ def test_get_transaction_receipt_for_mined_transaction(self, eth_tester):
         receipt = eth_tester.get_transaction_receipt(transaction_hash)
         assert receipt['transaction_hash'] == transaction_hash
 
-    def test_get_transaction_receipt_for_unmined_transaction(self, eth_tester):
+    def test_get_transaction_receipt_for_unmined_transaction_raises(self, eth_tester):
         eth_tester.disable_auto_mine_transactions()
         transaction_hash = eth_tester.send_transaction({
             "from": eth_tester.get_accounts()[0],
             "to": BURN_ADDRESS,
             "gas": 21000,
         })
-        receipt = eth_tester.get_transaction_receipt(transaction_hash)
```

Comment:
According to [eth_getTransactionReceipt ](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gettransactionreceipt) this shouldn't work.

Changing this test made me realise that get_transaction_by_hash should be able to retrieve the pending transaction - it doesn't right now.

Reply:
It does now